quickstart/daml/licensing-tests/daml/Licensing/Scripts/TestLicense.daml

```diff
@@ -0,0 +1,217 @@
+import qualified DA.Map as Map
+import DA.Date
+import DA.Time
+
+import Daml.Script
+
+
+import Splice.Amulet
+
+import Splice.Scripts.Util
+
+import Licensing.AppInstall
+import Licensing.License
+import Licensing.Util
+import Splice.Wallet.Payment
+import Splice.Scripts.TestWallet
+
+test : Script ()
+test = do
+  app@LicensingApp{..} <- setupLicensingApp
+  setupLicensingInstall app alice
+
+  -- start with 200 USD worth of CC
+  void $ tap amuletApp provider 200.0
+  checkProviderBalance app 200.0
+
+  -- setup license expiredAt (now = 2022-01-01T00:00:00Z)
+  licenseId <- setupLicense app alice
+  -- advance time to 2022-01-11T00:00:00Z (10 days) to simulate renewal base is right when renewed
+  passTime (days 10)
+
+  -- First renewal - Initial --
+  -- renew license for a year for fee of 20.0 USD worth of CC
+  (_, paymentRequestId)  <- renewLicense app licenseId 20.0 $ days 365
+
+  checkBalance app alice 1.0
+  amulet <- tap amuletApp alice 64.0
+
+  receipt <- acceptPaymentRequest amuletApp alice provider.primaryParty None paymentRequestId amulet
+
+  completeTransfer amuletApp provider licenseId receipt
+
+  (_, license) <- getLicense alice
+
+  unless (license.expiresAt == time (date 2023 Jan 11) 0 0 0) $
+    fail $ "Expected license to expire on 2023-01-11T00:00:00Z but got " <> show license.expiresAt
+
+  checkProviderBalance app 220.0
+  checkBalance app alice 45.0
+
+  -- Second renewal - Extension --
+  (licenseId, _) <- getLicense alice
+  -- advance time to 2022-07-01T00:00:00Z (171 days) to midyear sale and addition of extension to license
+  passTime (days 171)
+  -- renew license for another year this time for fee of 10.0 USD (midyear) worth of CC
+  (_, paymentRequestId)  <- renewLicense app licenseId 10.0 $ days 365
+
+  amulet <- tap amuletApp alice 50.0
+  receipt <- acceptPaymentRequest amuletApp alice provider.primaryParty None paymentRequestId amulet
+
+  completeTransfer amuletApp provider licenseId receipt
+  
+  checkProviderBalance app 230.0
+  checkBalance app alice 85.0
+
+  (_, license) <- getLicense alice
+
+  unless (license.expiresAt == time (date 2024 Jan 11) 0 0 0) $
+    fail $ "Expected license to expire on 2024-01-11T00:00:00Z but got " <> show license.expiresAt
+
+  -- Thrid renewal - Rejection --
+  (licenseId, _) <- getLicense alice
+  -- advance time to 2022-12-31T00:00:00Z (183 days) to end of year sale and rejection of payment
+  passTime (days 183)
+  -- renew license for another year this time for fee of 10.0 USD (end of year) worth of CC
+  (renewalRequestId, paymentRequestId)  <- renewLicense app licenseId 10.0 $ days 365
+  -- reject the request
+  rejectPaymentRequest app alice renewalRequestId paymentRequestId
+
+  -- ensure the payment and renewal request are archived
+  [] <- query @AppPaymentRequest alice.primaryParty
+  [] <- query @LicenseRenewalRequest alice.primaryParty
+
+  -- balance should remain the same
+  checkProviderBalance app 230.0
+  checkBalance app alice 85.0
+  
```

Comment:
TODO: 
- test concurrent renewals to see that the user gets two extensions
- test unhappy case where the payment request expires and we want to see that all of the involved contracts expire as well